cmake/module/TestAppendRequiredLibraries.cmake

```diff
@@ -25,10 +25,11 @@ function(test_append_socket_library target)
     }
   ")
 
-  include(CheckSourceCompilesAndLinks)
-  check_cxx_source_links("${check_socket_source}" IFADDR_LINKS_WITHOUT_LIBSOCKET)
+  include(CheckCXXSourceCompiles)
+  check_cxx_source_compiles("${check_socket_source}" IFADDR_LINKS_WITHOUT_LIBSOCKET)
   if(NOT IFADDR_LINKS_WITHOUT_LIBSOCKET)
-    check_cxx_source_links_with_libs(socket "${check_socket_source}" IFADDR_NEEDS_LINK_TO_LIBSOCKET)
+    set(CMAKE_REQUIRED_LIBRARIES socket)
```

Comment:
Could you add a `LINK_LIBRARIES` option to `check_cxx_source_compiles_with_flags` and use that here instead?

Reply:
Thanks! [Done](https://github.com/bitcoin/bitcoin/pull/31662#issuecomment-2593891189).